header/p2p/server.go

```diff
@@ -183,6 +184,10 @@ func (serv *ExchangeServer) handleRequest(from, to uint64) ([]*header.ExtendedHe
 	defer cancel()
 	headersByRange, err := serv.store.GetRangeByHeight(ctx, from, to)
 	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			log.Warnw("server: requested headers not found", "from", from, "to", to)
+			return nil, err
```

Comment:
We should return `header.ErrNotFound` here, actually, so that the logic above sends the correct status code to the remote side, otherwise, we just do a stream reset

Reply:
This is actually a bug and it very nice that we found it

Reply:
+1

Reply:
Nice catch. Thank you